src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -43,6 +43,11 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for saga storage. Select another persistence or disable the sagas feature using endpointConfiguration.DisableFeature<Sagas>()");
             }
 
+            if (!context.Container.HasComponent<ISagaIdGenerator>())
```

Comment:
Wouldn't it be very little code to make it explicit instead of having a black magic `setting.Set<ISagaIdGenerator>(new ServiceFabricAwareIdGenerator())`?

But I do see the YAGNI aspect to this so I'm fine either way